File: Backend/src/orm_classes/Device.py
from .shared import db


class Device(db.Model):
    id = db.Column(db.INTEGER, primary_key=True)
    devIP = db.Column(db.String(100), unique=False, nullable=False)
    name = db.Column(db.String(20), unique=False, nullable=False)
    serial = db.Column(db.String(20), unique=True, nullable=False)
    description = db.Column(db.String(1000), unique=False, nullable=True)
    state = db.Column(db.String(20), unique=False, nullable=False)
    puzzle = db.Column(db.INTEGER, db.ForeignKey('puzzle.id'), nullable=False)
    pubkey = db.Column(db.String(100), unique=False, nullable=False)
    node_state = db.Column(db.String(20), unique=False, nullable=False)
    is_event_device = db.Column(db.BOOLEAN)

    def __repr__(self):
        return 'ID: {}\nDevice IP: {}\nName: {}\nState: {}\nPuzzle: {}'\
            .format(self.id, self.devIP, self.name, self.serial, self.description, self.state, self.puzzle, self.node_state, self.is_event_device)

    def serialize(self):
        """Return object data in easily serializable format"""
        return {
            'id': self.id,
            'devIP': self.devIP,
            'name': self.name,
            'serial': self.serial,
            'description': self.description,
            'state': self.state,
            'puzzle': self.puzzle,
            'node_state': self.node_state,
            'is_event_device': self.is_event_device
        }

File: App/app/src/main/java/de/haw/riddle/ui/admin/model/Device.java
package de.haw.riddle.ui.admin.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class Device implements Parcelable {

    private String description;
    private long devIP;
    private final long id;
    @SerializedName("is_event_device")
    private boolean isEventDevice;
    private String name;
    @SerializedName("node_state")
    private String nodeState;
    @SerializedName("pubkey")
    private String publicKey;
    @SerializedName("puzzle")
    private long parentPuzzleId;
    private String serial;
    private String state;


    protected Device(Parcel in) {
        description = in.readString();
        devIP = in.readLong();
        id = in.readLong();
        isEventDevice = in.readByte() != 0;
        name = in.readString();
        nodeState = in.readString();
        publicKey = in.readString();
        parentPuzzleId = in.readLong();
        serial = in.readString();
        state = in.readString();
    }

    public static final Creator<Device> CREATOR = new Creator<Device>() {
        @Override
        public Device createFromParcel(Parcel in) {
            return new Device(in);
        }

        @Override
        public Device[] newArray(int size) {
            return new Device[size];
        }
    };

    public boolean isValid() {
        return name != null && !name.isEmpty();
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(description);
        parcel.writeLong(devIP);
        parcel.writeLong(id);
        parcel.writeByte((byte) (isEventDevice ? 1 : 0));
        parcel.writeString(name);
        parcel.writeString(nodeState);
        parcel.writeString(publicKey);
        parcel.writeLong(parentPuzzleId);
        parcel.writeString(serial);
        parcel.writeString(state);
    }
}
